contexts/arbo-context.tsx

```diff
@@ -47,51 +47,35 @@ function filterData(data: any[], filters: { [key: string]: string[] }): any[] {
     return filterKeys.every((key: string) => {
       if (!filters[key].length) return true;
 
-      if(key === "end_date") {
-        const filterEndDate = parseISO(filters["end_date"][0]);
+      if (key === "end_date") {
+        const filterEndDate = new Date(filters["end_date"][0]);
 
-        if(!filterEndDate) {
-          return true;
+        if (isNaN(filterEndDate.getTime())) {
+          return true; // Handle invalid date
         }
 
-        const itemDate = new Date(item.sample_end_date);
+        const itemStartDate = new Date(item.sample_start_date);
+        const itemEndDate = new Date(item.sample_end_date);
 
-        const filterEndUTC = Date.UTC(
-          filterEndDate.getUTCFullYear(),
-          filterEndDate.getUTCMonth(),
-        );
-
-        const itemDateUTC = Date.UTC(
-          itemDate.getUTCFullYear(),
-          itemDate.getUTCMonth(),
-        );
-        
-        return itemDateUTC <= filterEndUTC;
+        // Check for any overlap in the sampling period
+        return !(itemEndDate < filterEndDate && itemStartDate < filterEndDate);
```

Comment:
Are we sure about this logic?

`itemEndDate` is always after `itemStartDate` so if `filterEndDate` is bigger than `itemEndDate` it's automatically bigger than `itemStartDate` so we can simplify this to this:

`!(itemEndDate < filterEndDate && itemStartDate < filterEndDate)` -> `!(itemEndDate < filterEndDate)`

Which is the same as this: `itemEndDate > filterEndDate`
The condition we had before was `filterEndDate <= filterEndUTC`
So it looks to me like we've just inverted our logic here? I'm assuming the start date is the same thing.

I checked out your branch and tried using out the filter and it looks like I'm right that there's something wrong with the logic here. I gave a sampling start date of 1982 and a sampling end date of 2014 which should return a whole bunch of studies but it returned zero because I think it's looking for everything that started before 1982 and ended after 2014 instead of everything which started at some point between 1982 and 2014

![image](https://github.com/PathoTracker/Pathotracker/assets/86806388/08baf5ad-8584-4ca1-93ba-a533ae01f7ac)

Reply:
Will double check I had it traced out on paper but will double check. Yeah that is odd maybe it was doing the reverse. Weird i thought it was working for some of the date ranges I tried. This condition to get all the overlaps is kinda confusing haha. Will attempt to re-format it to get all the 4 conditions